spec/rspec/core/did_you_mean/proximity_spec.rb

```diff
@@ -0,0 +1,41 @@
+require 'tmpdir'
+require 'rspec/support/spec/in_sub_process'
+
+module RSpec
+  module Core
+    module DidYouMean
+      RSpec.describe Proximity do
+        # port of https://github.com/yuki24/did_you_mean/blob/master/test/edit_distance/jaro_winkler_test.rb
+        if RUBY_VERSION.to_f >= 2.0
```

Comment:
We have functions in RSpec::Support to take care of things like this, see how it does the specs for encoded strings.

Reply:
I looked at [encoded string spec](https://github.com/rspec/rspec-support/blob/master/spec/rspec/support/encoded_string_spec.rb) and it seems to use a similar approach (at line 88) as I am using. Am I looking in the right place?  

However, I do wonder about the exact implementation being used at line 88. In particular, using ```RUBY_VERSION < '2.1'``` appears to rely on a string comparison which can give incorrect results e.g '11' < '2.1' evaluates as true.

Reply:
The key difference is we use the function we need first, (for example line 43) then only use `RUBY_VERSION` for things we can't pick out by other means, we also gate this functionality out for the other gems via `https://github.com/rspec/rspec-support/blob/master/lib/rspec/support/ruby_features.rb`.

> However, I do wonder about the exact implementation being used at line 88. In particular, using `RUBY_VERSION < '2.1'` appears to rely on a string comparison which can give incorrect results e.g '11' < '2.1' evaluates as true.

It does, however `RUBY_VERSION` will always by `x.y.z` unless tampered with, and why we prefer feature detection instead.

Reply:
Thank you, I will have a further look.

Reply:
done, I think?